informational/req-res-reliability.md

```diff
@@ -0,0 +1,62 @@
+
+## Motivation
+
+Descriptions of mentioned protocols do not define some of the real world use cases that are oftenly observed in unreliable network environment. Such use cases can be: recovery from offline state, decrease rate of missed messages, increase probability of messages being broadcasted within the network.
+
+## Suggestions
+
+### Node health
+
+As a useful metric to define and implement for determining quality of provided service by a node:
+- unhealthy - no peer connections are available regardless of protocol;
+- minimally healthy:
+  - Relay has less than 4 peers connected;
+  - Filter and LightPush has one per each peer connection available;
+- sufficiently healthy:
+  - Relay has minimum 4 peers connected;
+  - more than 1 connection in Filter and at least 2 connections available in LightPush;
+
+### Peers and connection management
+
+- Each protocols should retain a pool of reliable peers. In case a protocol failed to use any peer more than once - connection to it should be dropped and new peer should be added to the pool instead. 
+
+- During discovery of new peers it is better to filter unwonted out based on ENR / multiaddress. For example in some cases `circuit-relay` addresses are not needed when we try to find and connect to peers directly.
+
+- When peer is discovered second time, we need to be sure to keep connection information up to date in Peer Store.
+
+### Light Push 
+
+- While sending message with Light Push - it is advised to use more than 1 peer in order to increase changes of delivering message.
+
+- If sending message is failed to all of the peers - node should try to re-send message after some interval and continue doing so while OK response is received. 
+
+### Filter
+
+- To decrease chances of missing messages a node can initiate more than one subscription through Filter protocol to the same content topic and filter out duplicates. This will increase bandwidth consumption and would depend on the information exchanged under content topic in use.
+
+- In case a node goes offline while having an active subscription - it is important to do ping again right after node appears online. In case ping fails - re-subscribe request should be fired to a new peer.
+
+- While registering Filter subscriptions - it is advised to batch requests for multiple content topics into one in order to reduce amount of queries sent to a node. 
+
+- During creation of a new subscription it can be benefitial to use only new peers to which no subscriptions yet present and not use peers with which Filter already failed.
+
+## Security/Privacy Considerations
+
+None of the mentioned recommendations incur privacy or security tradeoffs and in some cases increase k-anonymity (e.g having unique peers for Filter subscriptions).
+
```

Comment:
I don't see a section on regular store queries. But maybe that is already covered as part of relay reliability spec and we can just reference that section here indicating how it is used to identify missing messages (both sent and supposed to receive) and how fetch/resend is done to fill the gaps.

Reply:
added `Retry missing messages` section that covers regular store queries and approach described in here https://github.com/waku-org/specs/pull/18#pullrequestreview-2144776176